### PR TITLE
[REVIEW] Add notes in IO supported types doc table.

### PR DESCRIPTION
**Pull request by galipremsagar**
This PR adds a note for JSON writer, that it is not GPU accelerated. 
This PR also fixes a misc doc build issue around formatting.

### CODE REVIEW

**Comment by galipremsagar** [comment]
The page will look like this after these changes:

Screen Shot 2021-05-10 at 6 07 10 PM

File: docs/cudf/source/io-supported-types.rst [hunk 1]
```diff
@@ -5,58 +5,62 @@ The following table lists are compatible cudf types for each supported IO format
...
-    +-----------------------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+
-    | struct                | ❌     | ❌     | ✅     | ✅     | ❌     | ❌     | ❌     | ❌     | ❌     | ✅     | ✅     | ❌     | ❌     | ✅     | ✅     |
-    +-----------------------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+
-    | decimal64             | ❌     | ❌     | ✅     | ✅     | ❌     | ❌     | ❌     | ✅     | ❌     | ❌     | ❌     | ❌     | ❌     | ❌     | ❌     |
-    +-----------------------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+
+    +-----------------------+--------+--------+--------+--------+---------+--------+--------+--------+--------+-----------------+--------+--------+--------+--------+
+    |                       |       CSV       |      Parquet    |       JSON       |       ORC       |  AVRO  |       HDF       |       DLPack    |      Feather    |
```
Comment (kkraus14) [inline docs/cudf/source/io-supported-types.rst:11]: HDF reader / writer and Feather reader / writer aren't GPU accelerated either

Comment (vuule) [reply docs/cudf/source/io-supported-types.rst:11]: I think the note can be "Not GPU accelerated" so it can apply to any component.

Comment (galipremsagar) [reply docs/cudf/source/io-supported-types.rst]: Updated for HDF and feather too

**Review by kkraus14 — APPROVED** [review]

**Comment by kkraus14** [comment]
@gpucibot merge

**Pull request merged**